### PR TITLE
Support kubernetes.io~empty-dir mounts for WCOW

### PR DESCRIPTION
**Pull request by kevpar**
For LCOW, we special case mounts containing this string and set up a new
directory inside the UVM.

For WCOW, we can consider this as an
optimization in the future, but for now we will just treat empty-dir
mounts as a standard directory mount from the host. The reasoning for
this is twofold:
- Supporting a "sandbox://" mount in the UVM like we do for LCOW would
  require WCOW GCS changes, which are difficult since the WCOW GCS is a
  Windows component.
- Creating the directory inside of the UVM actually changes the
  mount semantics. For instance, something on the host may expect to be
  able to inspect the contents of the directory the empty-dir plugin has
  created. Additionally, the sandbox directory is expected to persist
  across reboots by default, and putting it in the UVM violates that.

Signed-off-by: Kevin Parsons <kevpar@microsoft.com>

### CODE REVIEW

File: pkg/server/container_create_windows.go [hunk 1]
```diff
@@ -368,11 +368,12 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
...
-			if sandboxPlatform != "linux/amd64" {
-				return nil, errors.Errorf(`kubernetes.io~empty-dir mounts are only supported for LCOW`, src)
-			}
-
+		} else if strings.Contains(src, "kubernetes.io~empty-dir") && sandboxPlatform == "linux/amd64" {
+			// kubernetes.io~empty-dir in the mount path indicates it comes from the kubernetes
+			// empty-dir plugin, which creates an empty scratch directory to be shared between
+			// containers in a pod. For LCOW, we special case this support and actually create
+			// our own directory inside the UVM. For WCOW, we want to skip this conditional branch
+			// entirely, and just treat the mount like a normal directory mount.
```
Comment (katiewasnothere) [inline pkg/server/container_create_windows.go:376]: Can you explain this more? In LCOW if we have this prefix we create a new directory in the sandbox's namespace called something like `sandboxMounts`. Do we need/want/have similar behavior here?

Comment (kevpar) [reply pkg/server/container_create_windows.go:376]: I think creating a directory in the UVM is a potential optimization in the future, but probably not something we want to do for now. I tried to address this in the PR description, can you see if that answers your question, or let me know how I can improve the wording? :)

Comment (dcantah) [reply pkg/server/container_create_windows.go:376]: We'd need windows GCS changes to mimic what we do for LCOW is probably the big deterrent but this is in the pr description. That fact could be added here to explain why we want to just treat it as a normal directory mount

**Comment by kevpar** [comment]
@katiewasnothere @dcantah Is this good to go in? Any other feedback?

**Review by katiewasnothere — APPROVED** [review]
LGTM

**Pull request merged**